AlicerceTests/Sources/Network/URLSessionNetworkStackTestCase.swift

```diff
@@ -0,0 +1,192 @@
+//
+//  URLSessionNetworkStackTestCase.swift
+//  Alicerce
+//
+//  Created by Luís Afonso on 07/04/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import XCTest
+
+@testable import Alicerce
+
+final class URLSessionNetworkStackTestCase: XCTestCase {
+
+    private let resource = Resource<Void>(path: "", method: .GET, parser: { _ in () })
+
+    // MARK: - Success tests
+
+    func testURLSession_WhenUsingConvenienceInit_ItShouldPopulateAllProperties() {
+        let url = URL(string: "http://0.0.0.0")!
+        let networkConfiguration = Network.Configuration(baseURL: url)
+
+        let urlSessionStack = Network.URLSessionNetworkStack(configuration: networkConfiguration)
+
+        // TODO: Replace with expectation
+        urlSessionStack.fetch(resource: resource) { (inner: () throws -> Data) in
+            do {
+                let _ = try inner()
+            } catch _ {}
+        }
+    }
+
+    func testURLSession_WhenResponseIsSuccessful_ItShouldCallCompletionClosureWithData() {
+        let baseURL = URL(string: "http://")!
+
+        let mockSession = MockURLSession()
```

Comment:
We could move this `mockSession` to a var on the test itself and reset it on each `setUp`/`tearDown`. It would avoid all this duplicated setup on multiple tests 😉